--- src/Terraria/Terraria/WindowsLaunch.cs
+++ src/tModLoader/Terraria/WindowsLaunch.cs
@@ -1,6 +_,8 @@
 #if WINDOWS
+using Microsoft.Xna.Framework;
 using System;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using System.Runtime.InteropServices;
 using Terraria.Social;
@@ -63,6 +_,20 @@
 					return Assembly.Load(array);
 				}
 			};
+
+			NativeLibrary.SetDllImportResolver(typeof(Game).Assembly, (name, assembly, searchPath) => {
+				try {
+					var files = Directory.GetFiles(Environment.CurrentDirectory, name + ".*");
+					var match = files.FirstOrDefault(s => File.Exists(s));
+					if (match != null && NativeLibrary.TryLoad(match, out var handle)) {
+						return handle;
+					}
+					return IntPtr.Zero;
+				}
+				catch (DirectoryNotFoundException e) {
+					throw new DirectoryNotFoundException("A needed library file was missing from the tModLoader/deps directory. " + e.Message, e);
+				}
+			});
 
 #if SERVER
 			WindowsLaunch._handleRoutine = new WindowsLaunch.HandlerRoutine(WindowsLaunch.ConsoleCtrlCheck);
